shell/platform/android/io/flutter/view/ResourceExtractor.java

```diff
@@ -76,23 +98,72 @@ protected Void doInBackground(Void... unused) {
 
             return null;
         }
+
+
+        /// Returns true if successfully unpacked APK resources,
+        /// otherwise deletes all resources and returns false.
+        private boolean extractAPK(@NonNull File dataDir) {
```

Comment:
Can this just be made static and kept in the outer class, instead of moving it here?

Reply:
It could, you just then have to pass more references around to the method here (`mResources`, `mAssetManager`, `mDataDirPath`)

Reply:
Nit - It might be slightly better, to keep it symmetrical with deleteFiles and checkTimestamp.

Reply:
LGTM